libsolidity/analysis/OverrideChecker.cpp

```diff
@@ -649,14 +649,19 @@ void OverrideChecker::checkAmbiguousOverrides(ContractDefinition const& _contrac
 			for (auto next = modifiers.upper_bound(*it); it != next; ++it)
 				baseModifiers.insert(*it);
 
-			checkAmbiguousOverridesInternal(std::move(baseModifiers), _contract.location());
+			checkAmbiguousOverridesInternal(std::move(baseModifiers), _contract.contractKind(), _contract.location());
 		}
 
 	}
 }
 
-void OverrideChecker::checkAmbiguousOverridesInternal(set<OverrideProxy> _baseCallables, SourceLocation const& _location) const
+void OverrideChecker::checkAmbiguousOverridesInternal(set<OverrideProxy> _baseCallables, ContractKind _contractKind, SourceLocation const& _location) const
 {
+	// Ambiguous overrides are acceptable within interfaces because the most-derived contract must eventually deal
```

Comment:
I don't think this will work like expected, because when doing the other checks, we sometimes rely on the fact that all other contracts have been checked already. What happens if you remove this check?

Reply:
Check re-added.